Make theme compatible with the latest 0.24 version

lizconlan opened the pull request:
Closes #92

lizconlan commented:
just this left to fix (I think)

![screen shot 2016-06-15 at 18 06 33](https://cloud.githubusercontent.com/assets/27760/16089641/02ada142-3324-11e6-8707-be00f466b536.png)

lizconlan commented:
New requests listing page:

![screen shot 2016-06-16 at 09 59 00](https://cloud.githubusercontent.com/assets/27760/16111212/706078b0-33a9-11e6-8dd5-d57665a60a30.png)

Old (current) requests listing page:

![screen shot 2016-06-16 at 10 02 57](https://cloud.githubusercontent.com/assets/27760/16111246/8f8f5378-33a9-11e6-80dc-e0766d3e63fd.png)

garethrees commented:
> just this left to fix (I think)

This is a problem in general (e.g. http://bilmehakki.org/list/successful)

garethrees commented on lib/views/general/_responsive_footer.html.erb:
```diff
@@ -24,7 +24,7 @@
         <li>
           <h4><%= _("A project by") %></h4>
-          <%= link_to image_tag('logo.png', :alt => "Diritto di Sapere logo"), "http://www.dirittodisapere.it/", :class => 'footer-logo' %>
+          <%= link_to image_tag('dds-logo.png', :alt => "Diritto di Sapere logo"), "http://www.dirittodisapere.it/", :class => 'footer-logo' %>
```
Change requested in https://github.com/mysociety/dirittodisapere-theme/issues/92#issuecomment-223601584

garethrees commented on lib/views/general/_responsive_topnav.html.erb:
```diff
@@ -14,7 +14,7 @@
+        <li><%= link_to "Chi Siamo", ' http://www.dirittodisapere.it/chi-siamo/' %></li>
         <li class="<%= 'selected' if params[:controller] == 'help' %>">
           <%= link_to _("Help"), help_about_path %>
         </li>
```
Search box looks broken:

![screen shot 2016-06-23 at 09 55 43](https://cloud.githubusercontent.com/assets/282788/16297412/abd4e0f0-3928-11e6-8dff-c1b21afa27e7.png)

Doesn't seem to be a consequence of updated (longer) translations (edited in web inspector).

Also, the search form method needs to be "GET" as per https://github.com/mysociety/alaveteli/blob/master/doc/CHANGES.md#upgrade-notes-11

lizconlan replied:
Ugh - CSS regression. Have already had to fix this for frontpage. Good catch, thanks!

lizconlan replied:
@lizconlan broken in Chrome and Firefox, you only fixed it for Safari

garethrees commented:
Notes for deployment:

- [ ] Make sure "new style" xapian compact script is used

garethrees commented:
- [ ] Ensure all overridden Ruby source files have encoding specifier. See 576b58803. (Note that we don't actually do this for whatdotheyknow-theme)

garethrees commented on lib/controller_patches.rb:
```diff
@@ -7,14 +7,8 @@
-          end
-      end
+    def frontpage
+      blog
```
Don't we want to set `@blog_items = []` if the feed times out?

lizconlan replied:
The [core controller blog method](https://github.com/mysociety/alaveteli/blob/0.24.1.7/app/controllers/general_controller.rb#L30-L56) handles the feed timeout and sets `@blog_items` for us

lizconlan replied:
...or should do.

garethrees replied:
Ah, yep. This seems to work okay.

lizconlan commented:
Closer. I think.

lizconlan commented:
Pretty sure this is working with core master now and that I've fixed the CSS gremlins (although the search box looks odd (too tall) and is forced onto its own line in IE8).

Points of concern/note:

- 47d00f1 is a temporary cheat (which will probably be reframed as a suggestion) that reverts a 2015 translation change which is in Transifex but isn't currently on the live site that breaks the navigation bar:
  - original (old) translation 
![screen shot 2016-06-29 at 12 18 49](https://cloud.githubusercontent.com/assets/27760/16450212/d693270a-3df3-11e6-952f-fc2b4c6128d3.png)
  - new (c. 2015) translation
![screen shot 2016-06-29 at 12 19 25](https://cloud.githubusercontent.com/assets/27760/16450222/e8597804-3df3-11e6-98a1-f78bb17a0424.png)
- I've changed the look of the search box (and chosen a variation of the standard green colour for the button that doesn't make my eyes hurt), hopefully this is ok. (Slightly different from out of date screen grabs, above)
- Updating the `request/new` view required some significant changes so hoping I haven't broken it. There are now fewer diffs between the overridden template and the one in core (note to self, raise bug against core for missed translation markup). Happy to split cb3c743 into whitespace vs functional changes if that would make it easier to review but I don't think it's too bad. Potential stumbling blocks:
  - Have I changed the original meaning/intent by moving the text around? (I think this is ok but would appreciate a second opinion)
  - In cb3c743, I've moved the text added as `request_advice` in 87b4033 (https://github.com/mysociety/dirittodisapere-theme/blob/master/lib/views/request/new.html.erb#L143-L151) to the `advice-panel` section and have opted to drop the first 2 lines in favour of the similar - but not identical - translation strings from Transifex https://github.com/mysociety/dirittodisapere-theme/blob/upgrade-to-0.24/lib/views/request/new.html.erb#L197-L209 which may not be entirely correct. 
- Have I overridden the css in the correct way (i.e. in the spirit of the modular system)?
- As the text colour used for the links is so dark that I've gone for `lighten` instead of `darken` as the hover/active style. Is this ok or is it jarring?

I think there should be some commit squashing/rebasing before this can be merged (e.g c3693b1 has been partially reverted in favour of newer commits) but reluctant to change the commit history mid review.

garethrees commented on locale-theme/it/app.po:
```diff
@@ -89,7 +89,7 @@ msgid "Information on emissions and discharges (e.g. noise, energy, radiation, w
 
 msgid "Make a request"
-msgstr ""
+msgstr "Inserisci richiesta"
```
When we pull from transifex this is going to get overridden. I suggest changing it to this in transifex, and when we email them, note that you've done so.

garethrees commented on lib/alavetelitheme.rb:
```diff
@@ -1,3 +1,4 @@
+# -*- encoding : utf-8 -*-
```
This misses a bunch of files:

- https://github.com/mysociety/dirittodisapere-theme/blob/6ed760cfc358c14a04f8bb499b3e50f6c7a1d56f/uninstall.rb
- https://github.com/mysociety/dirittodisapere-theme/blob/6ed760cfc358c14a04f8bb499b3e50f6c7a1d56f/post_install.rb

garethrees commented:
Superseded by https://github.com/mysociety/dirittodisapere-theme/pull/93
